packages/typespec-client-generator-core/src/validate.ts

```diff
@@ -16,12 +16,13 @@ import {
   UnionVariant,
 } from "@typespec/compiler";
 import { DuplicateTracker } from "@typespec/compiler/utils";
-import { getClientNameOverride } from "./decorators.js";
-import { AllScopes, clientNameKey, createTCGCContext, TCGCContext } from "./internal-utils.js";
+import { getClientNameOverride, createTCGCContext } from "./decorators.js";
+import { AllScopes, clientNameKey } from "./internal-utils.js";
 import { reportDiagnostic } from "./lib.js";
+import { TCGCContext } from "./interfaces.js";
 
 export function $onValidate(program: Program) {
-  const tcgcContext = createTCGCContext(program);
+  const tcgcContext = createTCGCContext(program, "@azure-tools/typespec-client-generator-core");
```

Comment:
I feel like it is good that this was not provided

Reply:
why would it be good if it wasn't applied?

Reply:
We are technically in a stage where there is no specific scope so this is kinda lying and might have unexpected consequences